main.go

```diff
@@ -138,19 +136,20 @@ func main() {
 	)
 
 	var g run.Group
+
 	m := metricshandler.New(
 		opts,
 		kubeClient,
 		storeBuilder,
 		opts.EnableGZIPEncoding,
 	)
-
 	// Run MetricsHandler
+	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
the initial background context should still be shared as the root of all context's .. it's also likely that we will make it cancelable in the future when we add listening to OS signals

Reply:
you also want to move this into the `{`, as that makes the ctx and cancel variables only accessible to the scope that run group is defined in, it prevents accidental shadowing that go attempts to do

Reply:
I'm unsure of what this means. Isn't this just another background context? It is only used once for the MetricsHandler.

The shared root is currently the main context if I understand correctly. Can't we just create a new context for the signal handler and use the Run Groups to cancel everything if such a situation happen?

Reply:
They are currently distinct background contexts. Typically they are wired up so they all lead to the root context, this is so that if a context somewhere in the middle gets cancelled, all children get cancelled. I added a review to show what I mean.

Reply:
Oh, that was a misunderstanding on my side. I thought that `context.Background()` was a singleton or something but it creates a different instance on each call.
I understand what you mean and will update with your patch.